core/src/main/scala/org/http4s/ServerSentEvent.scala

```diff
@@ -0,0 +1,78 @@
+package org.http4s
+
+import scala.util.Try
+import scalaz.stream.Process._
+import scalaz.stream.Process1
+import scalaz.stream.text.utf8Decode
+
+import scodec.bits.ByteVector
+
+case class ServerSentEvent(
+  data: String,
+  eventType: String = "message",
+  lastEventId: String = "",
```

Comment:
Two options here, neither totally appealing:
* make `eventType` and `lastEventId` optional to support deserialization, and lose inference of their default values.
* separate inbound and outbound models